fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/DimensionSearchServlet.java

```diff
@@ -0,0 +1,198 @@
+                    downloadFilename,
+                    perPage,
+                    page,
+                    dimensionDictionary
+            );
+
+            if (requestMapper != null) {
+                apiRequest = (DimensionsApiRequestImpl) requestMapper.apply(apiRequest, containerRequestContext);
+            }
+
+            PaginationParameters paginationParameters = apiRequest
+                    .getPaginationParameters()
+                    .orElse(ApiRequestImpl.DEFAULT_PAGINATION);
+
+            Pagination<DimensionRow> pagedRows;
+
+            SearchProvider searchProvider = apiRequest.getDimension().getSearchProvider();
+            if (searchProvider instanceof SearchQuerySearchProvider && !searchQuery.isEmpty()) {
+                // If requested dimension's search provider CAN support search queries, perform search query.
+                pagedRows = ((SearchQuerySearchProvider) searchProvider).findSearchRowsPaged(
+                        searchQuery,
+                        paginationParameters
+                );
+            } else {
+                // otherwise throw error indicating to not use the search query endpoint for this dimension.
+                // no message because catch clause will ignore it
+                throw new UnsupportedOperationException();
+            }
+
+            Response.ResponseBuilder builder = Response.status(Response.Status.OK);
+
+            Stream<Map<String, String>> rows = ResponsePaginator.paginate(builder, pagedRows, uriInfo)
+                    .map(DimensionRow::entrySet)
+                    .map(Set::stream)
+                    .map(stream ->
+                            stream.collect(
+                                    StreamUtils.toLinkedMap(
+                                            entry -> DimensionsServlet.getDescriptionKey(entry.getKey().getName()),
+                                            Map.Entry::getValue
+                                    )
+                            )
+                    );
+
+            Response response = formatResponse(
+                    apiRequest,
+                    builder,
+                    pagedRows,
+                    containerRequestContext,
+                    rows,
+                    UPDATED_METADATA_COLLECTION_NAMES.isOn() ? "dimensions" : "rows",
+                    null
+            );
+
+            LOG.debug("Dimension Value Endpoint Response: {}", response.getEntity());
+            return response;
+        } catch (UnsupportedOperationException e) {
+            String msg = String.format(
+                    "Dimension %s does not support search queries. Please try again using a standard api filter " +
+                            "against the dimension values endpoint.",
+                    apiRequest.getDimension()
+            );
+            LOG.debug(msg, e);
+            return Response.status(Response.Status.NOT_FOUND).entity(msg).build();
```

Comment:
I'm not sure 404 makes sense here. There's a difference between "We couldn't find anything that satifies your search query" and "we can't satisfy your search query." Maybe
405 Method Not Allowed makes more sense here? https://httpstatuses.com/405

Reply:
👍